### PR TITLE
[#9536] Improve E2E test stability and add test for searching response comments

### PR DESCRIPTION
**Pull request by wkurniawan07**
This will be an extended part of #9536 

- Resolved the high failure rate of Chrome-based E2E tests. It seems that a somewhat recent version of Chromedriver has a bug in setting page load timeout, so it is resolved by using native JS method. There may be other places where that bug occurs, but `driver.get` is the most obvious and by far most widely used call site.
- Separated `InstructorFeedbackResultsPageE2ETest` to multiple test cases. This is preferable because:
  - The entire test runs in 5.5 mins, so if anything breaks in the middle, everything has to be run all over again
  - The different view types of the results page technically create "different pages"
- Improved stability of `InstructorSearchPageE2ETest` sllightly
- Added some minor fixes to the search comments functionality: minor UI improvement and show participant comment on top of instructor comments
- Added E2E test portion for search response comments (another box ticked in the original issue!)

### CODE REVIEW

**Comment by rrtheonlyone** [comment]
Thanks for the improvements 👍 

Quick Question:

> It seems that a somewhat recent version of Chromedriver has a bug in setting page load timeout

What is your source for this? I could only find [this](https://bugs.chromium.org/p/chromedriver/issues/detail?id=907) but seems it has been patched already.

**Comment by wkurniawan07** [comment]
Running own trials. Just because it's not documented online doesn't mean it doesn't happen. The failures you see on Travis, they're very easily replicable offline.
And the Chrome tests you think are "stable" in GitHub actions, you need to be aware that they really are not.

**Comment by rrtheonlyone** [comment]
If you managed to do experimentation and show a more consistent test result, that is fine 👍 . However, it is also possible that there are other explanations for the behaviour rather than a bug in chrome-driver (depending on how you did the trials). 

> the Chrome tests you think are "stable" in GitHub actions, you need to be aware that they really are not.

I am not sure what you mean by this. I do not recall mentioning anything of that sort.

**Comment by wkurniawan07** [comment]
> it is also possible that there are other explanations for the behaviour rather than a bug in chrome-driver

If no Firefox failures are observed for the past 2-3 weeks and yet so many Chrome failures (I had to restart 50+ instances of Chrome tests in Travis, but never a single instance of Firefox tests), the conclusion is obvious. The only missing information is where the bug comes from, and that's where my own trials managed to uncover.

> I do not recall mentioning anything of that sort.

Maybe you didn't, but I want to prevent the mindset that moving to GitHub actions solve the problem of Chrome failures when it really didn't.

File: src/e2e/java/teammates/e2e/pageobjects/Browser.java [hunk 1]
```diff
@@ -131,6 +131,11 @@ public void close() {
      * Visits the given URL.
      */
     public void goToUrl(String url) {
+        if (TestProperties.BROWSER.equals(TestProperties.BROWSER_CHROME)) {
+            // Recent chromedriver has bug in setting page load timeout, which can potentially cause infinitely long waits
```
Comment (rrtheonlyone) [inline src/e2e/java/teammates/e2e/pageobjects/Browser.java:135]: what is the version of chromedriver + chrome that I can use to reproduce the said bug above? Might be good to mention that in this comment as well

**Review by rrtheonlyone — CHANGES_REQUESTED** [review]
Changes look good 👍 

Minor comments

File: src/e2e/java/teammates/e2e/cases/InstructorFeedbackResultsPageE2ETest.java [hunk 2]
```diff
@@ -249,11 +270,14 @@ private void testViews() {
 
         ______TS("GQR view: verify comments");
         resultsPage.verifyGqrViewComment(qn2, comment, responseWithComment, instructors, students, false);
+    }
 
-        // -------------------------------------- RQG View -------------------------------------- //
+    @Test
+    public void testRqgView() {
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/cases/InstructorFeedbackResultsPageE2ETest.java:276]: ? RQG and RGQ are two entirely different views

File: src/e2e/java/teammates/e2e/pageobjects/Browser.java (hunk 1 above)
Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/pageobjects/Browser.java:135]: I'm not really sure when the entire thing started, all I can tell is it is present in the latest.

File: src/e2e/java/teammates/e2e/pageobjects/InstructorSearchPage.java [hunk 3]
```diff
@@ -60,7 +69,15 @@ public void search(boolean searchForStudents, boolean searchForComments, String
             searchKeyword.clear();
             searchKeyword.sendKeys(searchTerm);
             click(searchButton);
-            waitForLoadingElement();
+            WebElement loadingContainer = null;
+            try {
+                loadingContainer = waitForElementPresence(By.className("loading-container"));
+            } catch (NoSuchElementException e) {
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/e2e/java/teammates/e2e/pageobjects/InstructorSearchPage.java:75]: Looks like you're right

File: src/e2e/java/teammates/e2e/cases/InstructorFeedbackResultsPageE2ETest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (rrtheonlyone) [reply src/e2e/java/teammates/e2e/cases/InstructorFeedbackResultsPageE2ETest.java:276]: oh woops sry, saw wrongly!

**Review by rrtheonlyone — APPROVED** [review]
LGTM 👍

**Pull request merged**